packages/gestalt/src/Text.js

```diff
@@ -91,26 +83,37 @@ export default function Text({
     underline && typography.underline,
     weight === 'bold' && typography.fontWeightBold,
     weight === 'normal' && typography.fontWeightNormal,
-    truncate && typography.truncate,
+    // `lineClamp` overrides `truncate`
+    truncate && !isNotNullish(lineClamp) && typography.truncate,
+    isNotNullish(lineClamp) && typography.lineClamp,
   );
+
   const Tag = inline ? 'span' : 'div';
 
   return (
     <Tag
       className={cs}
-      {...(truncate && typeof children === 'string' ? { title: children } : null)}
+      title={
+        (truncate || isNotNullish(lineClamp)) && typeof children === 'string' ? children : undefined
+      }
+      {...(lineClamp ? { style: { WebkitLineClamp: lineClamp } } : {})}
```

Comment:
Might be worth checking in with design. Would we want any constraints on how many lines we allow?

Reply:
Good question. I don't think that's a blocker for this PR, but should probably be resolved before we ship.